app/components/op_turbo_sandbox/projects/inline_edit_component.html.erb

```diff
@@ -0,0 +1,51 @@
+<%#-- copyright
+OpenProject is an open source project management software.
+Copyright (C) 2012-2023 the OpenProject GmbH
+
+This program is free software; you can redistribute it and/or
+modify it under the terms of the GNU General Public License version 3.
+
+OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+Copyright (C) 2006-2013 Jean-Philippe Lang
+Copyright (C) 2010-2013 the ChiliProject Team
+
+This program is free software; you can redistribute it and/or
+modify it under the terms of the GNU General Public License
+as published by the Free Software Foundation; either version 2
+of the License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program; if not, write to the Free Software
+Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+See COPYRIGHT and LICENSE files for more details.
+
+++#%>
+<%= component_wrapper(style: "margin-top: 5px;") do %>
+  <% if @state == :show %>
```

Comment:
I'm still wondering whether wrapping the state in the component itself will hurt us in the end, rather than wrapping the state in the state machine _before_ the view component. In more complex cases, you'll have a lot of branching that will complicate the view.

In this example, the branching happens only in the view. But imagine a more complex case, where methods on the VC would behave differently, you'd have branching there or start having methods like `<state>_link/name/etc` that simply hints at it being a different responsibility.

Reply:
I was thinking that we might make use of [variants](https://guides.rubyonrails.org/layouts_and_rendering.html#the-variants-option) in order to keep the different concerns related to a resource (show, edit, ...) inside the same component. But the controller would then still have to know [which state the component should be in to pass it in](https://github.com/opf/openproject/pull/13218/files#diff-6ea502f8b9172cbd8cf64e0be229eca0aae2bf18771e48fbfc02a311cc32045cR41) depending if rendering from a show or an edit action so it wouldn't help that much. And that example given for variants is actually a good one (desktop vs mobile). The variants pattern has its limits because we shouldn't nest deeply same as with any other inheritance. 

So from a controller level it is either
```
OpTurboSandbox::Projects::InlineEditComponent.new(project: @project, state: :edit)
```
and
```
OpTurboSandbox::Projects::InlineEditComponent.new(project: @project)
```

or 
```
OpTurboSandbox::Projects::InlineEditComponent.new(project: @project)
```
and
```
OpTurboSandbox::Projects::InlineShowComponent.new(project: @project)
```
 
Of those, I'd prefer the later.

We could also simplify this to 

```
OpTurboSandbox::Projects::EditComponent.new(project: @project, variant: :inline)
```
and
```
OpTurboSandbox::Projects::ShowComponent.new(project: @project, variant: inline)
```

as I would expect to have way more similarities between a show/edit in different contexts than between edit and show.

What happens inside the controllers is a different question. How many methods need to be shared between the two, then separate components? There are modules to take care of that but they as well do add a level of complexity as opposed to just have it all inside a file. On that it is probably personal preference. I'd go for including modules for sharing but I understand that people might have a different opinion on it.